### PR TITLE
Add error page to workspace creation

### PR DESCRIPTION
**Pull request by blrubenstein**
This adds an error to creating a workspace. It doesn't do any parsing on error code or text yet, but I don't know if we want to aim to add that now, or add it to the backlog as a non internal-launch ticket. (This time with rebasing)

### CODE REVIEW

File: ui/src/app/views/workspace-edit/component.ts [hunk 1]
```diff
@@ -56,9 +57,25 @@ export class WorkspaceEditComponent implements OnInit {
...
+              },
+              (error) => {
+                this.workspaceCreationError = true;
+              });
       }
     }
   }
+
+  navigateBack(): void {
+    this.router.navigate(['../..'], {relativeTo : this.route});
```
Comment (markfickett) [inline ui/src/app/views/workspace-edit/component.ts]: I think the Location service would be better here; for now everyone will get to the new-workspace page from the same place, but this should future-proof it without much effort (if it works as advertized): https://stackoverflow.com/questions/35446955/how-to-go-back-last-page-in-angular-2 .

File: ui/src/app/views/workspace-edit/component.ts [hunk 2]
```diff
@@ -56,9 +57,25 @@ export class WorkspaceEditComponent implements OnInit {
       } else {
         this.buttonClicked = true;
         this.valueNotEntered = false;
-        this.errorHandlingService.retryApi(this.workspacesService.createWorkspace(this.workspace))
-            .subscribe(cohorts => this.router.navigate(['../..'], {relativeTo : this.route}));
+        this.errorHandlingService.retryApi(
+          this.workspacesService.createWorkspace(this.workspace))
+            .subscribe(
+              (workspace) => {
```
Comment (markfickett) [inline ui/src/app/views/workspace-edit/component.ts]: I guess `cohorts` before was a typo (which we didn't use anyway)?

File: ui/src/app/views/workspace-edit/component.html [hunk 3]
```diff
@@ -102,3 +102,11 @@
     <p *ngIf="this.valueNotEntered;">* is a required field</p>
   </div>
 </div>
+<clr-modal [(clrModalOpen)]="workspaceCreationError">
+  <h3 class="modal-title">Error:</h3>
+  <div class="modal-body">Could not create workspace.</div>
+  <div class="modal-footer">
+    <button type="button" class="btn btn-primary" (click)="navigateBack()">Cancel Creation</button>
+    <button type="button" class="btn btn-primary" (click)="resetWorkspaceCreation()">Try Again</button>
```
Comment (markfickett) [inline ui/src/app/views/workspace-edit/component.html]: What do you think of "Keep Editing"? I would expect "Try Again" to do basically what `retryApi` does -- just keep issuing RPCs without my intervention.

Comment (blrubenstein) [reply ui/src/app/views/workspace-edit/component.html]: I agree, I think "Keep Editing" is better. Making the change.

File: ui/src/app/views/workspace-edit/component.ts (hunk 2 above)
Comment (blrubenstein) [reply ui/src/app/views/workspace-edit/component.ts]: It was (I think a copy/paste remnant.) I'm actually going to see if I can subscribe with no arguments, even if the call returns something.

File: ui/src/app/views/workspace-edit/component.ts (hunk 1 above)
Comment (blrubenstein) [reply ui/src/app/views/workspace-edit/component.ts]: I'll take a look at this!

**Pull request merged**